Content.Client/Inventory/ClientInventorySystem.cs

```diff
@@ -25,39 +57,276 @@ public override void Initialize()
+            component.AttachedToGameHud = false;
+        }
+
+        private void OnShutdown(EntityUid uid, ClientInventoryComponent component, ComponentShutdown args)
+        {
+            OnPlayerDetached(uid, component);
+        }
+
+        private void OnPlayerAttached(EntityUid uid, ClientInventoryComponent component, PlayerAttachedEvent args)
+        {
+            if(component.AttachedToGameHud) return;
+
+            _gameHud.InventoryButtonVisible = true;
+            _gameHud.BottomLeftInventoryQuickButtonContainer.AddChild(component.BottomLeftButtons);
+            _gameHud.BottomRightInventoryQuickButtonContainer.AddChild(component.BottomRightButtons);
+            _gameHud.TopInventoryQuickButtonContainer.AddChild(component.TopQuickButtons);
+            component.AttachedToGameHud = true;
+        }
+
+        private void UpdateHudTheme(int obj)
         {
-            foreach (var (_, ent) in component.AllSlots)
+            if (!_gameHud.ValidateHudTheme(obj))
             {
-                if (ent != default)
+                return;
+            }
+
+            foreach (var inventoryComponent in EntityManager.EntityQuery<ClientInventoryComponent>(true))
+            {
+                foreach (var slotButton in inventoryComponent.SlotButtons)
                 {
-                    RaiseLocalEvent(ent, args, false);
+                    foreach (var btn in slotButton.Value)
+                    {
+                        btn.RefreshTextures(_gameHud);
+                    }
                 }
             }
         }
 
         public override void Shutdown()
         {
             CommandBinds.Unregister<ClientInventorySystem>();
+            _config.UnsubValueChanged(CCVars.HudTheme, UpdateHudTheme);
             base.Shutdown();
         }
 
+        private void OnInit(EntityUid uid, ClientInventoryComponent component, ComponentInit args)
+        {
+            _clothingSystem.InitClothing(uid, component);
+
+            if (!TryGetUIElements(uid, out var window, out var bottomLeft, out var bottomRight, out var topQuick,
+                    component))
+                return;
+
+            component.InventoryWindow = window;
+            component.BottomLeftButtons = bottomLeft;
+            component.BottomRightButtons = bottomRight;
+            component.TopQuickButtons = topQuick;
+        }
+
+        private void HoverInSlotButton(EntityUid uid, string slot, ItemSlotButton button, InventoryComponent? inventoryComponent = null, SharedHandsComponent? hands = null)
+        {
+            if (!Resolve(uid, ref inventoryComponent))
+                return;
+
+            if (!Resolve(uid, ref hands, false))
+                return;
+
+            if (!hands.TryGetActiveHeldEntity(out var heldEntity))
+                return;
+
+            _itemSlotManager.HoverInSlot(button, heldEntity,
+                CanEquip(uid, heldEntity, slot, out _, inventory: inventoryComponent));
+        }
```

Comment:
Not super important, but currently this creates a green ghost when you hover over an already occupied slot, although clicking on it does not swap out the equipped clothing.

Maybe either allow swapping, or make the ghost red / not drawn when the slot is occupied?

Reply:
partially addressed. i wont resolve this due to not having fixed the x swap hands ghost update issue yet

Reply:
actually, the swap hands is for the next person tbh. it needs an event that tells me what hand is currently active